src/PointwiseFunctions/GeneralRelativity/ComputeGhQuantities.cpp

```diff
@@ -209,6 +215,222 @@ Scalar<DataType> time_deriv_of_lapse(
   return dt_lapse;
 }
 
+template <size_t SpatialDim, typename Frame, typename DataType>
+void spatial_deriv_of_shift(
+    const gsl::not_null<tnsr::iJ<DataType, SpatialDim, Frame>*> deriv_shift,
+    const Scalar<DataType>& lapse,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi) noexcept {
+  for (size_t i = 0; i < SpatialDim; ++i) {
+    for (size_t j = 0; j < SpatialDim; ++j) {
+      for (size_t a = 0; a < SpatialDim + 1; ++a) {
+        for (size_t b = 0; b < SpatialDim + 1; ++b) {
+          deriv_shift->get(i,j) += (inverse_spacetime_metric.get(j+1, a)
+                + spacetime_unit_normal.get(j+1) * spacetime_unit_normal.get(a))
+            * spacetime_unit_normal.get(b) * phi.get(i, a, b);
+        }
+      }
+      deriv_shift->get(i,j) *= get(lapse);
+    }
+  }
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::iJ<DataType, SpatialDim, Frame> spatial_deriv_of_shift(
+    const Scalar<DataType>& lapse,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_unit_normal,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi) noexcept {
+  auto deriv_shift =
```

Comment:
default construct, as it will be resized and initialized in the not-null function

Reply:
changed